### ISSUE TITLE
How to improve the synthesis speed on CPU

### ISSUE THREAD

**Issue by 1105060120** [fatchord/WaveRNN#102]
Hello,everyone.I synthesis the model on CPU for 6 minutes. How can I speed up and do real time synthesis on cpu.thank you

**Comment by oytunturk** [comment]
Tacotron and WaveRNN won't work real-time on CPU. You may want to look into LPCNet as an alternate to WaveRNN. However, you'll still need to use an inferior model instead of Tacotron to predict acoustic features from text/linguistic features such as simpler feedforward or recursive deepnets.

**Comment by 1105060120** [comment]
@oytunturk .But this paper from Google says it supports real-time on CPU,and i saw somebody export to C++ and real time inference on CPU

**Comment by 1105060120** [comment]
@oytunturk  do you know how to export this to C++,thanks

**Comment by oytunturk** [comment]
Which Google paper are you refering to?

**Comment by 1105060120** [comment]
@oytunturk  waveRNN

**Comment by oytunturk** [comment]
That paper only discusses the vocoder portion and, yes, with the sparse
wavernn model which does heavy weight pruning, it’s possible to run the
sample generation from already predicted spectrograms. How are you planning
to generate the spectrogram from text on a CPU? Tacotron won’t work that
fast. Also, the tricks they implemented for fast wavernn inference are not
available in this repo (as far as I know).

**Comment by 1105060120** [comment]
@oytunturk  maybe i use the fast speech to  generate the spectrograms.The tricks for WaveRNN is not available?

**Comment by oytunturk** [comment]
Yes, you’ll need a spectrogram generator + a neural vocoder that are both
significantly faster than real-time on a CPU. I’d look into models that can
be parallelized to use multi-threading. I guess there will be significant
quality loss wrto Tacotron+Wavenet/WaveRNN baselines.

**Comment by TheButlah** [comment]
@1105060120 For what its worth, 6 minutes seems really slow. Although its not realtime, my 2016 macbook pro synthesises at about a fifth of realtime speed (not including vocoder), and Tacotron is not particularly slow.

I'm sure that if you used weight pruning and did some optimizations you could get it to work acceptably. Its sure as hell a lot better than 15 minutes to synthesize an utterance on a 2080 Ti using WaveNet

**Comment by 1105060120** [comment]
i had improved the speed of WaveRNN,thank you.@TheButlah

**Comment by OliverMathias** [comment]
@1105060120 how did you go about improving the speed? I'm trying to process longer texts and it's becoming quite time-consuming.

**Comment by 1105060120** [comment]
@OliverMathias  weight pruning and C++ infer.

**Comment by muntasir2000** [comment]
> @OliverMathias weight pruning and C++ infer.

Can you share some more details on how you did weight pruning and C++ infer?

**Comment by tsungruihon** [comment]
@1105060120 Would you mind sharing the inference speed on 1s audio ?
